### ISSUE TITLE
[clang-tidy] False negative readability-redundant-parentheses member of struct

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#186749]
Example:

```cpp
struct Foo
{
    bool x;
};

int main()
{
    bool a{};
    Foo b{};

    if ((a))  // OK, warns here
    {}

    if ((b.x))  // NOK, does not warn here!
    {}
}
```
This warns about the first `if`, but not about the second. I think the second one is a false negative and should warn about it too.

https://godbolt.org/z/sMn8M7ezb
